pkg/hooks/gcr/hook_test.go

```diff
@@ -0,0 +1,77 @@
+package gcr
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/gitops-tools/image-updater/pkg/hooks"
+)
+
+var _ hooks.PushEvent = (*PushMessage)(nil)
+var _ hooks.PushEventParser = Parse
+
+func TestParse(t *testing.T) {
+	req := makeHookRequest(t, "testdata/push_event.json")
+
+	hook, err := Parse(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &PushMessage{
+		Action: "INSERT",
+		Digest: "gcr.io/mynamespace/repository@sha256:6ec128e26cd5",
+		Tag:    "gcr.io/mynamespace/repository:latest",
+	}
+	if diff := cmp.Diff(want, hook); diff != "" {
+		t.Fatalf("hook doesn't match:\n%s", diff)
+	}
+}
+
+func TestPushedImageURL(t *testing.T) {
+	hook := &PushMessage{
+		Action: "INSERT",
+		Digest: "gcr.io/mynamespace/repository@sha256:6ec128e26cd5",
+		Tag:    "gcr.io/mynamespace/repository:latest",
+	}
+	want := "gcr.io/mynamespace/repository:latest"
+
+	if u := hook.PushedImageURL(); u != want {
+		t.Fatalf("got %s, want %s", u, want)
+	}
+}
+
+func TestRepository(t *testing.T) {
+	hook := &PushMessage{
+		Action: "INSERT",
+		Digest: "gcr.io/mynamespace/repository@sha256:6ec128e26cd5",
+		Tag:    "gcr.io/mynamespace/repository:latest",
+	}
+	want := "gcr.io/mynamespace/repository"
+
+	if u := hook.EventRepository(); u != want {
+		t.Fatalf("got %s, want %s", u, want)
+	}
+}
+
+func makeHookRequest(t *testing.T, fixture string) []byte {
```

Comment:
This could become `makeHookBody` ? or perhaps even `readFixture` now that it's just reading a file into bytes?